avocado/utils/process.py

```diff
@@ -1641,3 +1641,21 @@ def get_owner_id(pid):
         return os.stat('/proc/%d/' % pid).st_uid
     except OSError:
         return None
+
+
+def get_perf_events(pattern):
+    """
+    Run 'perf list' command, with the matching pattern create a
+    list and return it.
+
+    :param pattern: Pattern to search.
+    :type pattern: str.
+
+    :return: list of events matching the 'pattern'.
+    :rtype: list of str.
+    """
+    list_of_events = []
+    for line in run("perf list").stdout.split('\n'):
```

Comment:
using `system_output` function will good as it already return `stdout` of command

Reply:
@PraveenPenguin Thanks for your feedback.
To get the required output I have to do the following,

`with run`
```
for line in process.run("perf list").stdout.split('\n'):
```

`with system_output`
```
for line in process.system_output("perf list").split('\n'):
```

I don't see much difference between those two call outs.

Reply:
@nasastry yes both will does some that's why saying `system_output` can be used